File: 0933-number-of-recent-calls/0933-number-of-recent-calls.py
class RecentCounter:

    def __init__(self):
        self.dq = deque()

    def ping(self, t: int) -> int:
        self.dq.append(t)
        
        while self.dq[0] < self.dq[-1] - 3000:
            self.dq.popleft()
            
        return len(self.dq)


# Your RecentCounter object will be instantiated and called as such:
# obj = RecentCounter()
# param_1 = obj.ping(t)

File: 0933-number-of-recent-calls/0933-number-of-recent-calls.cpp
class RecentCounter {
private:
    queue<int> mq;
    
public:
    RecentCounter() {
        
    }
    
    int ping(int t) {
        mq.push(t);
        while (mq.front() < t - 3000) {
            mq.pop();
        }
        return mq.size();
    }
};

/**
 * Your RecentCounter object will be instantiated and called as such:
 * RecentCounter* obj = new RecentCounter();
 * int param_1 = obj->ping(t);
 */